src/ucp/core/ucp_context.c

```diff
@@ -22,6 +23,7 @@ ucp_am_handler_t ucp_am_handlers[UCP_AM_ID_LAST] = {{0, NULL, NULL}};
 static const char *ucp_atomic_modes[] = {
     [UCP_ATOMIC_MODE_CPU]    = "cpu",
     [UCP_ATOMIC_MODE_DEVICE] = "device",
+    [UCP_ATOMIC_MODE_GUESS]  = "guess",
```

Comment:
:) I was thinking about this. 
Auto - sounds that we guarantee best possible tuning for all possible configurations, which is not truee

Guess - sounds like we are trying to come with a reasonable tuning, even so it may not work.